packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/raw/RawManifest.kt

```diff
@@ -72,104 +72,101 @@ abstract class RawManifest(protected val json: JSONObject) {
 
   abstract fun getAssets(): JSONArray?
 
+  abstract fun getExpoGoConfigRootObject(): JSONObject?
+  abstract fun getExpoClientConfigRootObject(): JSONObject?
+
   fun isDevelopmentMode(): Boolean {
+    val expoGoRootObject = getExpoGoConfigRootObject() ?: return false
     return try {
-      json.has("developer") &&
-          json.has("packagerOpts") &&
-          json.getJSONObject("packagerOpts").optBoolean("dev", false)
+      expoGoRootObject.has("developer") &&
+              expoGoRootObject.has("packagerOpts") &&
+              expoGoRootObject.getJSONObject("packagerOpts").optBoolean("dev", false)
     } catch (e: JSONException) {
       false
     }
   }
 
   fun isDevelopmentSilentLaunch(): Boolean {
+    val expoGoRootObject = getExpoGoConfigRootObject() ?: return false
     return try {
-      json.has("developmentClient") &&
-          json.getJSONObject("developmentClient")
-            .optBoolean("silentLaunch", false)
+      expoGoRootObject.has("developmentClient") &&
+              expoGoRootObject.getJSONObject("developmentClient").optBoolean("silentLaunch", false)
     } catch (e: JSONException) {
       false
     }
   }
 
   fun isUsingDeveloperTool(): Boolean {
+    val expoGoRootObject = getExpoGoConfigRootObject() ?: return false
     return try {
-      json.has("developer") && json.getJSONObject("developer").has("tool")
+      expoGoRootObject.has("developer") && expoGoRootObject.getJSONObject(
+        "developer"
+      ).has("tool")
     } catch (e: JSONException) {
       false
     }
   }
 
-  fun getSlug(): String? = if (json.has("slug")) {
-    json.optString("slug")
-  } else {
-    null
-  }
+  abstract fun getSlug(): String?
 
-  fun getDebuggerHost(): String = json.optString("debuggerHost")
-  fun getMainModuleName(): String = json.optString("mainModuleName")
+  fun getDebuggerHost(): String = getExpoGoConfigRootObject()!!.getString("debuggerHost")
+  fun getMainModuleName(): String = getExpoGoConfigRootObject()!!.getString("mainModuleName")
```

Comment:
Are we sure we want to switch from the optional `optString` to the strict `getString` here?

Seems more correct, but will it break things?

Reply:
The only callsite is in a naked try/catch so I think it should be fine. It seems like they're expected to throw if the string isn't defined.